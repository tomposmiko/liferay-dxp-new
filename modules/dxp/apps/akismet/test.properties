##
## Test Batch
##

    #
    # Relevant
    #

    test.batch.run.property.query[functional-tomcat*-mysql*-jdk8][relevant]=\
        (testray.main.component.name ~ "Akismet")

##
## Testray
##

    testray.main.component.name=Akismet