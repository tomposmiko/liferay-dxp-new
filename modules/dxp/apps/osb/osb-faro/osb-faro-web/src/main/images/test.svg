<svg fill="none" height="20" viewBox="0 0 20 20" width="20" xmlns="http://www.w3.org/2000/svg">
	<rect fill="#6B6C7E" height="20" rx="5" width="20" />
	<path d="M14.7653 13.6478C13.308 10.8283 11.8837 9.2493 11.8762 7.75V5.5C11.8762 5.5 12.1619 5.5007 12.2512 5.5C13.2184 5.4918 13.2937 4.0382 12.2512 4H7.75116C6.72389 4.05789 6.77545 5.46133 7.75116 5.5H8.12616V7.75C8.13975 9.25164 6.56006 11.0352 5.23702 13.6478C4.92225 14.2696 5.09452 15.8924 6.99202 16H13.0103C14.6774 15.9386 15.1429 14.3784 14.7653 13.6478ZM12.9285 14.5H7.05577C6.84577 14.5 6.57342 14.2649 6.74077 13.9075L7.37499 12.6252C9.27108 12.6252 9.97022 11.1231 11.5103 11.1241C11.5103 11.1241 13.1483 13.784 13.2058 13.8625C13.4507 14.1979 13.2137 14.5 12.9285 14.5Z" fill="white" />
</svg>