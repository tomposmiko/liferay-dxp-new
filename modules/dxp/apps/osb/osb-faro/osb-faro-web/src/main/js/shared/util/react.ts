import {isEqual} from 'lodash';

export const getDisplayName = WrappedComponent =>
	WrappedComponent.displayName || WrappedComponent.name || 'Component';

type HasChanges = <T>(
	prev: T | object,
	next: T | object,
	...keys: string[]
) => boolean;

/**
 * Compare previous state or props object by provided keys to detect changes.
 */
export const hasChanges: HasChanges = (prev = {}, next = {}, ...keys) => {
	for (const key of keys) {
		if (key in next) {
			const newVal = next[key];

			const prevVal = prev[key];

			if (!isEqual(newVal, prevVal)) {
				return true;
			}
		}
	}

	return false;
};
