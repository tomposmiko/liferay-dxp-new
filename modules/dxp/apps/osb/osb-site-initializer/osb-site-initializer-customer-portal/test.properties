##
## Test Batch
##

    #
    # Relevant
    #

    test.batch.run.property.query[functional-tomcat*-mysql*-jdk8][relevant]=\
        (portal.acceptance == true) AND \
        (\
            (testray.component.names ~ "OSB Site Initializer Customer Portal") OR \
            (testray.main.component.name ~ "OSB Site Initializer Customer Portal")\
        )