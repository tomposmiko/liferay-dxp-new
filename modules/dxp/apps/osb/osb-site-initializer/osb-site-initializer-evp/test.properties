##
## Test Batch
##

    #
    # Relevant
    #

    test.batch.run.property.query[functional-tomcat*-mysql*-jdk8][relevant]=\
        (portal.acceptance == true) AND \
        (\
            (testray.component.names ~ "OSB Site Initializer EVP") OR \
            (testray.main.component.name ~ "OSB Site Initializer EVP")\
        )

##
## Testray
##

    testray.main.component.name=OSB Site Initializer EVP