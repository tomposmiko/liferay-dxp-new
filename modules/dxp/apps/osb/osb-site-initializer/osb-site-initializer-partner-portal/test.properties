##
## Salesforce
##

    #site.initializer.partner.portal.login.url=
    #site.initializer.partner.portal.consumer.key=
    #site.initializer.partner.portal.consumer.secret=
    #site.initializer.partner.portal.username=
    #site.initializer.partner.portal.password=

##
## Test Batch
##

    #
    # Relevant
    #

    test.batch.run.property.query[functional-tomcat*-mysql*-jdk8][relevant]=\
        (portal.acceptance == true) AND \
        (\
            (testray.component.names ~ "OSB Site Initializer Partner Portal") OR \
            (testray.main.component.name ~ "OSB Site Initializer Partner Portal")\
        )

##
## Testray
##

    testray.main.component.name=OSB Site Initializer Partner Portal