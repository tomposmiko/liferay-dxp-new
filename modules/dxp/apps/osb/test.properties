##
## Testray
##

    testray.main.component.name=Customer Portal