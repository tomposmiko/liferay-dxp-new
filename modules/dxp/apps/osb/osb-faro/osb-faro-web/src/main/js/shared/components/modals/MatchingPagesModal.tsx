import ClayButton from '@clayui/button';
import MetadataTag from 'settings/recommendations/components/MetadataTag';
import Modal from 'shared/components/modal';
import React from 'react';
import RecommendationPageAssetsQuery from 'settings/recommendations/queries/RecommendationPageAssetsQuery';
import {
	createOrderIOMap,
	getSortFromOrderIOMap,
	TITLE
} from 'shared/util/pagination';
import {
	EXCLUDE,
	Filter,
	getFilterValueBreakdown
} from 'settings/recommendations/utils/utils';
import {getMapResultToProps} from 'shared/hoc/mappers/metrics';
import {graphql} from '@apollo/react-hoc';
import {isArray, isString} from 'lodash';
import {omit} from 'lodash';
import {OrderedMap} from 'immutable';
import {OrderParams} from 'shared/util/records';
import {withBaseResults, withStatefulPagination} from 'shared/hoc';

const withData = () =>
	graphql(RecommendationPageAssetsQuery, {
		options: ({
			delta,
			itemFilters,
			orderIOMap,
			page,
			query,
			useNegateValue
		}: {
			delta: number;
			itemFilters: Filter[];
			orderIOMap: OrderedMap<string, OrderParams>;
			page: number;
			query: string;
			useNegateValue: boolean;
		}) => ({
			fetchPolicy: 'no-cache',
			variables: {
				keywords: query,
				propertyFilters: itemFilters.map(({name, value}) => ({
					filter: value,
					negate: useNegateValue ? name === EXCLUDE : false
				})),
				size: delta,
				sort: getSortFromOrderIOMap(orderIOMap),
				start: (page - 1) * delta
			}
		}),
		props: getMapResultToProps(({pageAssets: {pageAssets, total}}) => ({
			items: pageAssets,
			total
		}))
	});

const TableWithData = withStatefulPagination(
	withBaseResults(withData, {
		emptyDescription: Liferay.Language.get(
			'there-were-no-matching-items-for-this-string'
		),
		emptyTitle: Liferay.Language.get('no-matches-found'),
		getColumns: ({secondColumnHeader}) => [
			{
				accessor: 'title',
				className: 'table-cell-expand text-truncate',
				label: Liferay.Language.get('page-name')
			},
			{
				accessor: secondColumnHeader || 'url',
				className: 'secondary-info table-cell-expand text-truncate',
				dataFormatter: val => {
					if (isString(val)) {
						return val;
					} else if (isArray(val)) {
						return val.map(({value}) => value).join(', ');
					}
				},
				label: secondColumnHeader || 'url',
				sortable: false
			}
		],
		showDropdownRangeKey: false
	}),
	{
		initialDelta: 10,
		initialOrderIOMap: createOrderIOMap(TITLE)
	},
	props => omit(props, 'onSearchValueChange'),
	false
);

interface IMatchingPagesModalProps {
	itemFilters: Filter[];
	onClose: () => void;
	useNegateValue: boolean;
}

const MatchingPagesModal: React.FC<IMatchingPagesModalProps> = ({
	itemFilters,
	onClose,
	useNegateValue = false
}) => {
	const {name, value} = itemFilters[0];

	const {exactMatchSign, metadataTag, rule} = getFilterValueBreakdown(value);

	const customFilter = itemFilters.length === 1 && metadataTag;

	return (
		<Modal className='matching-pages-modal-root' size='xl'>
			<Modal.Header
				onClose={onClose}
				title={Liferay.Language.get('matching-pages')}
			/>

			<Modal.Body>
				{!!customFilter && !useNegateValue && (
					<div>
						<span className='include-exclude'>
							{`${
								name === EXCLUDE
									? Liferay.Language.get('exclude')
									: Liferay.Language.get('include')
							}:`}
						</span>

						<MetadataTag value={metadataTag} />

						<span className='rule'>
							{exactMatchSign ? `"${rule}"` : rule}
						</span>
					</div>
				)}
			</Modal.Body>

			<TableWithData
				itemFilters={itemFilters}
				secondColumnHeader={customFilter}
				useNegateValue={useNegateValue}
			/>

			<Modal.Footer>
				<ClayButton
					className='button-root'
					displayType='primary'
					onClick={onClose}
				>
					{Liferay.Language.get('done')}
				</ClayButton>
			</Modal.Footer>
		</Modal>
	);
};

export default MatchingPagesModal;
