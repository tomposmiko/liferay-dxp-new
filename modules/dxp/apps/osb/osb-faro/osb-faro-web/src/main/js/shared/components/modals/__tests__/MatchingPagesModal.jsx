import MatchingPagesModal from '../MatchingPagesModal';
import mockStore from 'test/mock-store';
import React from 'react';
import {MockedProvider} from '@apollo/react-testing';
import {mockRecommendationPageAssetsReq} from 'test/graphql-data';
import {Provider} from 'react-redux';
import {render} from '@testing-library/react';
import {StaticRouter} from 'react-router-dom';
import {waitForLoading} from 'test/helpers';

jest.unmock('react-dom');

describe('MatchingPagesModal', () => {
	it('should render', async () => {
		const {container} = render(
			<StaticRouter>
				<MockedProvider mocks={[mockRecommendationPageAssetsReq([])]}>
					<Provider store={mockStore()}>
						<MatchingPagesModal
							itemFilters={[
								{
									name: 'includeFilter',
									value: '.*custom-assets'
								}
							]}
						/>
					</Provider>
				</MockedProvider>
			</StaticRouter>
		);

		await waitForLoading(container);

		jest.runAllTimers();

		expect(container).toMatchSnapshot();
	});
});
