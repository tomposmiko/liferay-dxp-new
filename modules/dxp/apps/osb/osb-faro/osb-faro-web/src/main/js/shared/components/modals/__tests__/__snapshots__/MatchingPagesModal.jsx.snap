// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchingPagesModal should render 1`] = `
<div>
  <div
    aria-modal="true"
    class="modal-dialog matching-pages-modal-root modal-xl"
    role="dialog"
  >
    <div
      class="modal-content"
    >
      <div
        class="modal-header"
      >
        <h4
          class="modal-title"
        >
          Matching Pages
        </h4>
      </div>
      <div
        class="modal-body"
      />
      <div
        class="d-flex flex-column flex-grow-1 justify-content-between"
      >
        <div
          class="toolbar-root"
        >
          <nav
            class="navbar navbar-root management-bar management-bar-light navbar-expand"
          >
            <div
              class="container-fluid"
            >
              <ul
                class="nav-root front-nav navbar-nav"
                role="tablist"
              />
              <div
                class="navbar-form navbar-form-autofit"
              >
                <div
                  class="input-group search"
                >
                  <div
                    class="input-group-item"
                  >
                    <input
                      class="input-root form-control input-group-inset input-group-inset-after"
                      placeholder="Search"
                      type="text"
                      value=""
                    />
                    <div
                      class="input-group-inset-item input-group-inset-item-after"
                    >
                      <button
                        class="button-root btn btn-unstyled"
                        type="button"
                      >
                        <svg
                          class="icon-root lexicon-icon lexicon-icon-search"
                          viewBox=""
                        >
                          <use
                            xlink:href="/o/osb-faro-web/dist/sprite.svg#search"
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </nav>
        </div>
        <div
          class="error-display-root flex-grow-1 error-spacer no-results-root flex-grow-1"
        >
          <div
            class="no-results-content"
          >
            <h4
              class="no-results-title"
            >
              An unexpected error occurred.
            </h4>
            <button
              class="button-root btn btn-secondary"
              type="button"
            >
              Reload
            </button>
          </div>
        </div>
      </div>
      <div
        class="modal-footer"
      >
        <button
          class="button-root btn btn-primary"
          type="button"
        >
          Done
        </button>
      </div>
    </div>
  </div>
</div>
`;
